### PR TITLE
Added new toolbar which contains the cut, copy and paste

### PR DESCRIPTION
**Pull request by ksoze95**
Resolves: #3693 

Provide cut, copy and paste functionality from a toolbar.

This PR have the following couple of issues.

1. The paste toolbar button is enabled on the application starts up, it should be disabled until a copy action has first been preformed.
2. The new toolbar should be hidden by default.  When ToolManager.cpp was modified to hide the toolbar, audio setup and both meters toolbars are moved to above all the other toolbars.
 

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-theme/AllThemeResources.h
Comment (Paul-Licameli) [inline libraries/lib-theme/AllThemeResources.h:117]: After this is merged, remember to tell @LWinterberg to update the "sprite sheet" in documentation for users who customize the theme.

File: src/toolbars/CutCopyPasteToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/CutCopyPasteToolBar.cpp:90]: For the fourth argument -- an internal identifier the user doesn't see, which may be used as a key in the configuration file -- do not use the '/' character.  I think it might get misinterpreted as multiple config file path components.  Just omit those characters.

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]

File: src/toolbars/CutCopyPasteToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/CutCopyPasteToolBar.cpp:140]: Make a literal instead of repeating the string constant

File: src/toolbars/CutCopyPasteToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/CutCopyPasteToolBar.cpp:28]: This entire file is a cut-and-paste of EditToolBar with some modifications.

These two beg to be unified into a common base class, simply parameterized by a table of buttons and commands.

Such a simple, table driven toolbar might even become a useful toolkit neutral facade if it is done right.

None of this comment is a request for changes now.

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.cpp:1]: This new source file should have comments at top about its purpos

File: src/toolbars/CutCopyPasteToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/CutCopyPasteToolBar.cpp:170]: I suggest an OrderingHint to control the position of insertion of this menu item.  Did design specify it?  Maybe it should come after the edit toolbar, not at the bottom of the menu.

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.cpp:61]: Two integers are passed into this function, just so they can be added.

Why not simplify, passing in just the sum?

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.cpp:26]: I prefer member initializations where possible instead of mutations inside the constructor body.

But beware, it should be
`mButtons(size)`

not the curly-bracket form!  Because `vector` has constructors taking `initializer_list` and you don't want overload resolution to one of those.

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.cpp:24]: The sequence of intializers should match the sequence of member declarations, because it is the latter that really determines the sequence of execution.

It is unimportant in this case but I think this is one of the warnings we should elevate to errors in our build.  In some cases there may be surprises.

File: src/toolbars/EditToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/EditToolBar.cpp:108]: Please avoid unnecessary movements of functions and tables within files.  It makes review more difficult.  You could instead leave them where they were, so that the diff contains fewer lines.

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [reply src/toolbars/ToolBarButtons.cpp:61]: OK, a later commit changes this

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.cpp:49]: "command"

File: src/toolbars/EditToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/EditToolBar.cpp:176]: I like the effort to unify repepetious code into a utility class.

It might go farther (in 3.3 development).

The `Populate()` routines might also become completely table-driven if `Entry` could indicate a separator and also what the initial enabled state should be.

The `_EVENT_TABLE` macros might be removed and instead `ToolBarButtons` could `Bind` an event handler to each button as it is made.  (The preferred thing in newer wxWidgets code)

It might even be not too difficult to devise a platform-neutral facade class to unify these two simple toolbars, and also use a registry of button entries so that an add-on might insert buttons.

File: src/toolbars/ToolBarButtons.h
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.h:23]: I don't like a simple bland name like `Entry` in the global namespace.  This might be a nested type in `ToolBarButtons`.

**Comment by Paul-Licameli** [comment]
No more to say in this round of review.

File: src/toolbars/ToolBarButtons.cpp
Comment (crsib) [reply src/toolbars/ToolBarButtons.cpp:1]: I think we should stop that practice and stick just to SPDX license identifiers

File: src/toolbars/ToolBarButtons.cpp
Comment (crsib) [reply src/toolbars/ToolBarButtons.cpp:61]: I am not quite sure what do you mean

Comment (Paul-Licameli) [reply src/toolbars/ToolBarButtons.cpp:61]: I was reviewing the commits sequentially.  My comment applies to a step of the code transformation that is not the last.  The final version uses only one id argument.

File: src/toolbars/ToolBarButtons.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBarButtons.cpp:60]: Still misspelled

File: src/toolbars/ToolManager.cpp
Comment (crsib) [reply src/toolbars/ToolManager.cpp:678]: This is required, so we can properly setup the whole chain of neighbors.

File: src/toolbars/ToolBar.cpp
Comment (Paul-Licameli) [inline src/toolbars/ToolBar.cpp:653]: You rewrote my proposal with more generality.  Very well.

But as you might guess, I don't really like `DefaultConfigEntry` because it is a failure of open-endedness of the set of toolbars.  I would rather take the time fo make a new Registry that scatters the definition of the default configuration.

Each toolbar should decide where to insert itself.

For that, each toolbar's .cpp file would need to refer to another toolbar, but only the weak for of dependency that mentions a textual id, and if the toolbar module isn't present, there is no harm.

My pull request Open-ended-toolbars does what is needed to eliminate the centralized enumeration and identify toolbars by strings instead.  It doesn't yet go further to build up the default configuration by collecting registry items.

**Review by Paul-Licameli — APPROVED** [review]

**Pull request merged**
